qa/common/infer_util.py

```diff
@@ -84,474 +90,309 @@ def infer_exact(tester, pf, tensor_shape, batch_size,
     # class outputs the result value/probability is returned as a
     # float so must use fp32 range in that case.
     rinput_dtype = _range_repr_dtype(input_dtype)
-    routput0_dtype = _range_repr_dtype(output0_dtype if output0_raw else np.float32)
-    routput1_dtype = _range_repr_dtype(output1_dtype if output1_raw else np.float32)
+    routput0_dtype = _range_repr_dtype(
+        output0_dtype if output0_raw else np.float32)
+    routput1_dtype = _range_repr_dtype(
+        output1_dtype if output1_raw else np.float32)
     val_min = max(np.iinfo(rinput_dtype).min,
-                np.iinfo(routput0_dtype).min,
-                np.iinfo(routput1_dtype).min) / 2
+                  np.iinfo(routput0_dtype).min,
+                  np.iinfo(routput1_dtype).min) / 2
     val_max = min(np.iinfo(rinput_dtype).max,
-                np.iinfo(routput0_dtype).max,
-                np.iinfo(routput1_dtype).max) / 2
+                  np.iinfo(routput0_dtype).max,
+                  np.iinfo(routput1_dtype).max) / 2
 
     num_classes = 3
 
-    input0_list = list()
-    input1_list = list()
-    expected0_list = list()
-    expected1_list = list()
-    expected0_val_list = list()
-    expected1_val_list = list()
-    for b in range(batch_size):
-        in0 = np.random.randint(low=val_min, high=val_max,
-                                size=tensor_shape, dtype=rinput_dtype)
-        in1 = np.random.randint(low=val_min, high=val_max,
-                                size=tensor_shape, dtype=rinput_dtype)
-        if input_dtype != np.object:
-            in0 = in0.astype(input_dtype)
-            in1 = in1.astype(input_dtype)
-
-        if not swap:
-            op0 = in0 + in1
-            op1 = in0 - in1
-        else:
-            op0 = in0 - in1
-            op1 = in0 + in1
-
-        expected0_val_list.append(op0)
-        expected1_val_list.append(op1)
-        if output0_dtype == np.object:
-            expected0_list.append(np.array([unicode(str(x), encoding='utf-8')
-                                            for x in (op0.flatten())], dtype=object).reshape(op0.shape))
-        else:
-            expected0_list.append(op0.astype(output0_dtype))
-        if output1_dtype == np.object:
-            expected1_list.append(np.array([unicode(str(x), encoding='utf-8')
-                                            for x in (op1.flatten())], dtype=object).reshape(op1.shape))
-        else:
-            expected1_list.append(op1.astype(output1_dtype))
-
-        if input_dtype == np.object:
-            in0n = np.array([str(x) for x in in0.reshape(in0.size)], dtype=object)
-            in0 = in0n.reshape(in0.shape)
-            in1n = np.array([str(x) for x in in1.reshape(in1.size)], dtype=object)
-            in1 = in1n.reshape(in1.shape)
+    if (pf == "batch_to_nobatch") or (pf == "nobatch_to_batch") or (pf == "mix_nobatch_batch"):
```

Comment:
There seem to be a lot of changes here in input data preparation. Is this all due to how inputs are now a single numpy array for the entire batch?

Reply:
Why not making `infer_exact()` accept the actual shape as the parameter, then it is the caller's responsibility to provide proper shape, and it is more intuitive in respect to v2 spec.

Reply:
Done